### PR TITLE
Dev 2

### PR DESCRIPTION
**Pull request by DZMITRYSHAUTSOU**

### CODE REVIEW

File: dev-2/dev-2/StringAnalyzer.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            Sequence = sequence;
+        }
+        /// <summary>
+        /// This method calls every other method of the class
+        /// </summary>
+        public void Analyze()
+        {
+            LongestUniqueSymbolSubstring = GetUniqueSymbolSequence();
+            LongestRepetitiveDigitSubstring = GetRepetitiveDigitSequence();
+            LongestRepetitiveLetterSubstring = GetRepetitiveLetterSequence();
```
Comment (IgorKheidorov) [inline dev-2/dev-2/StringAnalyzer.cs:30]: Better to call Display separately, not in Analyze method

File: dev-2/dev-2/StringAnalyzer.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        public string Sequence { get; } = String.Empty;
+        private string LongestUniqueSymbolSubstring { get; set; } = String.Empty;
+        private string LongestRepetitiveDigitSubstring { get; set; } = String.Empty;
+        private string LongestRepetitiveLetterSubstring { get; set; } = String.Empty;
+        /// <summary>
+        /// Sets readonly propertie used for analyzing
+        /// </summary>
+        /// <param name="sequence">String for analyzing</param>
+        public StringAnalyzer(string sequence)
+        {
```
Comment (IgorKheidorov) [inline dev-2/dev-2/StringAnalyzer.cs:20]: Check sequence for null

File: dev-2/dev-2/StringAnalyzer.cs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            DisplayInfo();
+        }
+        /// <summary>
+        /// Finds unique symbol sequence
+        /// </summary>
+        /// <returns>string thar contains longest unique symbol sequence</returns>
+        public string GetUniqueSymbolSequence()
+        {
+            for (int i = Sequence.Length; i > 0; i--)
+            {
```
Comment (IgorKheidorov) [inline dev-2/dev-2/StringAnalyzer.cs:40]: Better Sequence.Length -i

File: dev-2/dev-2.tests/StringAnalyzerTests.cs [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+            //Act
+            string result = analyzer.GetUniqueSymbolSequence();
+
+            //Assert
+            Assert.AreEqual(expected, result);
+        }
+#endregion
+
+        #region LongestRepetitiveDigitSubstringMethod
+        [TestMethod]
```
Comment (IgorKheidorov) [inline dev-2/dev-2.tests/StringAnalyzerTests.cs:116]: Use more common names for test methods- negative, positive, Better LongestRepetitiveDigitSubstringFromLettersStringNegative- do not use details of test in the name - string like "aaaa" in a name not a good idea

File: dev-2/dev-2.tests/StringAnalyzerTests.cs [hunk 5]
```diff
@@ -0,0 +1,209 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace dev_2.tests
+{
+    [TestClass]
+    public class StringAnalyzerTests
+    {
+        #region RepetitiveOnlyMethod
+        [TestMethod]
```
Comment (IgorKheidorov) [inline dev-2/dev-2.tests/StringAnalyzerTests.cs:9]: The same as for 116

File: dev-2/dev-2.tests/StringAnalyzerTests.cs [hunk 6]
```diff
@@ -0,0 +1,209 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace dev_2.tests
+{
```
Comment (IgorKheidorov) [inline dev-2/dev-2.tests/StringAnalyzerTests.cs:4]: You do not create test for null string, I am sure that we will have exception

File: dev-2/dev-2.tests/StringAnalyzerTests.cs [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+            Assert.AreEqual(expected, result);
+        }
+
+        [TestMethod]
+        public void LongestUniqueSubstring_abcd_result_abcd()
+        {
+            //Arrange
+            string testString = "abcd";
+            string expected = "abcd";
+            StringAnalyzer analyzer = new StringAnalyzer(testString);
```
Comment (IgorKheidorov) [inline dev-2/dev-2.tests/StringAnalyzerTests.cs:105]: can simplify as  Assert.AreEqual((new StringAnalyzer(testString)).GetUniqueSymbolSequence(), result);

**Review by IgorKheidorov — APPROVED** [review]
